File: core/src/main/java/net/particify/arsnova/core/web/exceptions/NotFoundException.java
package net.particify.arsnova.core.web.exceptions;

/**
 * Not Found means status code 404.
 */
public class NotFoundException extends RuntimeException {
  private static final long serialVersionUID = 1L;

  public NotFoundException() {
    super();
  }

  public NotFoundException(final String message) {
    super(message);
  }

  public NotFoundException(final Throwable e) {
    super(e);
  }

  public NotFoundException(final String message, final Throwable e) {
    super(message, e);
  }
}

File: authz/src/main/kotlin/net/particify/arsnova/authz/exception/NotFoundException.kt
package net.particify.arsnova.authz.exception

import org.springframework.http.HttpStatus
import org.springframework.web.server.ResponseStatusException

class NotFoundException : ResponseStatusException {
  constructor() : super(HttpStatus.NOT_FOUND)
  constructor(reason: String?) : super(HttpStatus.NOT_FOUND, reason)
  constructor(reason: String?, cause: Throwable?) : super(HttpStatus.NOT_FOUND, reason, cause)
}
